source/logic.rst

```diff
@@ -0,0 +1,492 @@
+then the absolute value of ``x * y`` is less than ``ε``."
+In Lean, the implication arrows associate to the right,
+making it easy to chain implications in this way.
+
+You have already seen that even though the universal quantifier
+in this statement
+ranges over objects and the implication arrows introduce hypotheses,
+Lean treats the two in very similar ways.
+In particular, if you have proved a theorem of that form,
+you can apply it to objects and hypotheses in the same way:
+
+.. code-block:: lean
+
+    import data.real.basic
+
+    -- BEGIN
+    lemma my_lemma : ∀ x y ε : ℝ,
+      0 < ε → ε ≤ 1 → abs x < ε → abs y < ε → abs (x * y) < ε :=
+    sorry
+
+    section
+      variables a b δ : ℝ
+      variables (h₀ : 0 < δ) (h₁ : δ ≤ 1)
+      variables (ha : abs a < δ) (hb : abs b < δ)
+
+      #check my_lemma a b δ
+      #check my_lemma a b δ h₀ h₁
+      #check my_lemma a b δ h₀ h₁ ha hb
+    end
+    -- END
+
+You have also already seen that it is common in Lean
+to use curly brackets to make quantified variables implicit
+when they can be inferred from subsequent hypotheses.
+When we do that, we can just apply a lemma to the hypotheses without
+mentioning the objects.
+
+.. code-block:: lean
+
+    import data.real.basic
+
+    -- BEGIN
+    lemma my_lemma : ∀ {x y ε : ℝ},
+      0 < ε → ε ≤ 1 → abs x < ε → abs y < ε → abs (x * y) < ε :=
+    sorry
+
+    section
+      variables a b δ : ℝ
+      variables (h₀ : 0 < δ) (h₁ : δ ≤ 1)
+      variables (ha : abs a < δ) (hb : abs b < δ)
+
+      #check my_lemma h₀ h₁ ha hb
+    end
+    -- END
+
+At this stage, you also know that if you use
+the ``apply`` tactic to apply ``my_lemma``
+to a goal of the form ``abs (a * b) < δ``,
+you are left with new goals that require you to  prove
+each of the hypotheses.
+
+To prove a statement like this, use the ``intros`` tactic.
+Take a look at what it does in this example:
+
+.. code-block:: lean
+
+    import data.real.basic
+
+    -- BEGIN
+    lemma my_lemma : ∀ {x y ε : ℝ},
+      0 < ε → ε ≤ 1 → abs x < ε → abs y < ε → abs (x * y) < ε :=
+    begin
+      intros x y ε epos ele1 xlt ylt,
+      sorry
+    end
+    -- END
+
+We can use any names we want for the universally quantified variables;
+they do not have to be ``x``, ``y``, and ``ε``.
+(If there is only one variable, you can use the singular ``intro``
+instead of ``intros``.)
+After the ``intros`` command,
+the goal is what it would have been at the start if we
+listed all the variables and hypotheses *before* the colon,
+as we did in the last section.
+In a moment, we will see why it is sometimes necessary to
+introduce variables and hypotheses after the proof begins.
+
+To help you prove the lemma, we will start you off:
+
+.. code-block:: lean
+
+    import data.real.basic
+
+    -- BEGIN
+    lemma my_lemma : ∀ {x y ε : ℝ},
+      0 < ε → ε ≤ 1 → abs x < ε → abs y < ε → abs (x * y) < ε :=
+    begin
+      intros x y ε epos ele1 xlt ylt,
+      suffices h : abs x * abs y < 1 * ε,
```

Comment:
I don't think this is a great example. The intermediate statement is not something we would state on paper, and there is no issue using `calc` from start to finish. I'm sure we can find more natural places to introduce `suffices`.